### PR TITLE
Creates Hardened AMI promotion tool

### PR DESCRIPTION
**Pull request by tcsc**
Creates a tool that handles AMI promotion by finding the latest AMI for a given Teleport release and makes it as public.

This patch also moves the `update-ami-id` script into its own package directory in the `assets/aws` module, in order to both highlight its use and tidy up the module root.

### CODE REVIEW

File: assets/aws/cmd/make-public/main.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+			{Name: aws.String("tag:TeleportVersion"), Values: []string{teleportVersion}},
+			{Name: aws.String("tag:TeleportEdition"), Values: []string{edition}},
+			{Name: aws.String("tag:TeleportFipsEnabled"), Values: []string{fips}},
+			{Name: aws.String("tag:BuildType"), Values: []string{"production"}},
+		},
+		Owners: []string{accountId},
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("Failed to query AMIs: %w", err)
```
Comment (strideynet) [inline assets/aws/cmd/make-public/main.go:134]: nit:
```suggestion
		return "", fmt.Errorf("querying AMIs: %w", err)
```

File: assets/aws/cmd/make-public/main.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+	// I'm assuming that we will have few enough images returned in
+	// any given search thatits not worth setting up a fancy sorting
+	// predicate
+	newestTimestamp := time.Time{}
+	newestAMI := -1
+	for i := range resp.Images {
+		creationDate, err := time.Parse(time.RFC3339, *resp.Images[i].CreationDate)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse timestamp %w", err)
```
Comment (strideynet) [inline assets/aws/cmd/make-public/main.go:149]: nit:
```suggestion
			return "", fmt.Errorf("parsing timestamp: %w", err)
```

File: assets/aws/cmd/make-public/main.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+				if err != nil {
+					log.Printf("WARNING: Failed to make ami %q public: %s", ami, err)
+					continue
+				}
+			}
+		}
+	}
+}
+
+var notFound error = fmt.Errorf("Not Found")
```
Comment (strideynet) [inline assets/aws/cmd/make-public/main.go:119]: nit:
```suggestion
var notFound error = fmt.Errorf("not found")
```

**Review by strideynet — APPROVED** [review]
Just some nits re: error strings - otherwise looks good.

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**